accepted/0042-isolated-mode.md

```diff
@@ -0,0 +1,347 @@
+# Isolated mode
+
+## Summary
+
+This RFC is a proposal to add a new opt-in installation mode.
+
+This mode aims at preventing workspaces from over-sharing their dependencies. With this mode, two workspaces will share a dependency only if both have declared it in their package.json.
+
+This RFC aims at improving projects with workspaces, but the suggested implementation will also happen to bring value to projects not using workspaces. Because `isolated-mode` is valuable for any project, it will also be available for projects not using workspaces.
+
+## Motivation
+
+The introduction of workspaces in npm v7 brought the ability to split large code-bases into smaller pieces of code with declared dependencies between them.
+
+Several build tools took advantage of these declared dependencies to split the builds into multiple steps (one per workspace). These build tools take advantage of the declared dependencies between workspaces to only rebuild the workspaces that have not changed since the last build. With the current way npm works, any dependency change to any workspace may affect any other workspace. This means that the build systems need to rebuild the entire project every time a single workspace modifies a dependency. This is inefficient and does not scale.
+
```

Comment:
We should mention that workspaces in npm v7 facilitate the "monorepo" pattern (since you mention it later).

The motivations are:
* monorepo performance
* monorepo dependency isolation (prevent side effects when changing dependencies of one workspace to another)

I think this could be a bit more direct and clearer in this section specifically.